src/persist/src/indexed/runtime.rs

```diff
@@ -0,0 +1,233 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! WIP
+
+#![allow(missing_docs)]
+
+use std::sync::mpsc::{Receiver, RecvError, SendError};
+use std::sync::{mpsc, Arc, Mutex};
+use std::thread::{self, JoinHandle};
+
+use log;
+
+use crate::error::Error;
+use crate::indexed::{Indexed, IndexedSnapshot};
+use crate::storage::{Blob, Buffer};
+use crate::Id;
+
+enum Cmd {
+    Register(Id, Box<dyn FnOnce(Result<(), Error>) + Send + 'static>),
+    Write(
+        Id,
+        Vec<((String, String), u64, isize)>,
+        Box<dyn FnOnce(Result<(), Error>) + Send + 'static>,
+    ),
+    Seal(Id, u64, Box<dyn FnOnce(Result<(), Error>) + Send + 'static>),
+    Snapshot(
+        Id,
+        Box<dyn FnOnce(Result<IndexedSnapshot, Error>) + Send + 'static>,
+    ),
+    Stop,
+}
+
+#[derive(Clone)]
+pub struct RuntimeCore {
+    handle: Arc<Mutex<Option<JoinHandle<()>>>>,
```

Comment:
this is semantically more of a one shot channel. the awkward second Option (the first being `core: Option<RuntimeCore>` in Runtime, which feels good) is here because of the semantics mismatch

Reply:
I guess maybe I don't understand why `RuntimeCore` needs to be separated from `Runtime`!